pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -41,9 +50,114 @@ func NewPlugin() *Plugin {
 	return &Plugin{}
 }
 
+// RequireTLS returns true and pointer to mTLS params if the policy use mTLS for (peer) authentication.
+// (note that mTLS params can still be nil). Otherwise, return (false, nil).
+func RequireTLS(policy *authn.Policy) (bool, *authn.MutualTls) {
+	if policy == nil {
+		return false, nil
+	}
+	if len(policy.Peers) > 0 {
+		for _, method := range policy.Peers {
+			switch method.GetParams().(type) {
+			case *authn.PeerAuthenticationMethod_Mtls:
+				return true, method.GetMtls()
+			default:
+				continue
+			}
+		}
+	}
+	return false, nil
+}
+
+// JwksURIClusterName returns cluster name for the jwks URI. This should be used
+// to override the name for outbound cluster that are added for Jwks URI so that they
+// can be referred correctly in the JWT filter config.
+func JwksURIClusterName(hostname string, port *model.Port) string {
+	const clusterPrefix = "jwks."
+	const maxClusterNameLength = 189 - len(clusterPrefix)
+	name := hostname + "|" + port.Name
+	if len(name) > maxClusterNameLength {
+		prefix := name[:maxClusterNameLength-sha1.Size*2]
+		sum := sha1.Sum([]byte(name))
+		name = fmt.Sprintf("%s%x", prefix, sum)
```

Comment:
I felt it will be easier to debug if we choose the cluster name as

hostname + port.name

Not to SHA it.

Reply:
This follow the way naming cluster else where. The cluster name is used if shorter than max length, else it chopped, and replace with SHA (you still see some of its prefix)

Also note, for v2, this might be not used, as we will inject the external service, and let's pilot create cluster accordingly (i.e we don't control how to name it)